tests/tracking/test_fluent.py

```diff
@@ -681,3 +683,98 @@ def test_delete_tag():
     with pytest.raises(MlflowException):
         mlflow.delete_tag("b")
     mlflow.end_run()
+
+
+@pytest.fixture
+def library_to_mlflow_module():
+    return {
+        "tensorflow": "tensorflow",
+        "keras": "keras",
+        "fastai": "fastai",
+        "pyspark": "spark",
+        "sklearn": "sklearn",
+        "xgboost": "xgboost",
+        "lightgbm": "lightgbm",
+        "mxnet.gluon": "gluon",
+    }
+
+
+def clear_imports_and_import_hooks():
+    import wrapt
+
+    for integration_name in library_to_mlflow_module.keys():
+        try:
+            del sys.modules[integration_name]
+        except Exception:
+            pass
+        try:
+            del wrapt.importer._post_import_hooks[integration_name]
+        except Exception:
+            pass
+
+
+@pytest.mark.large
+def test_universal_autolog_does_not_throw_if_specific_autolog_throws(
+    mocker, library_to_mlflow_module,
+):
+    clear_imports_and_import_hooks()
```

Comment:
Instead of adding this before and after each test, can we define it as a fixture and auto-use it before each test case? see https://docs.pytest.org/en/stable/fixture.html#autouse-fixtures-xunit-setup-on-steroids